orc8r/cloud/go/swagger/combine.go

```diff
@@ -0,0 +1,215 @@
+ limitations under the License.
+*/
+
+package swagger
+
+import (
+	"regexp"
+	"sort"
+
+	"magma/orc8r/cloud/go/tools/swaggergen/generate"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+)
+
+
+// CombineStr is a wrapper function around the function Combine()
+// which serves to combine the contents of multiple swagger specs.
+// The output is the merged copy of the swagger spec
+func CombineStr(common string, configs []string) (string, error, error) {
+	cfgs, commonCfg, err := ConvertToSwagger(common, configs)
+	if err != nil {
+		return "", err, nil
+	}
+	
+	combinedSpec, warnings := Combine(commonCfg, cfgs)
+
+	out, err := marshalToYAML(combinedSpec)
+	if err != nil {
+		return "", err, nil
+	}
+
+	return out, nil, warnings
+}
+
+// Combine multiple Swagger specs, giving precedence to the "common" spec.
+// Returned "error" contains warnings for any overwritten fields.
+//
+// This custom-built functionality mirrors the "official" implementation.
+// See: https://github.com/go-openapi/analysis/blob/master/mixin.go
+func Combine(common generate.SwaggerSpec, specs []generate.SwaggerSpec) (generate.SwaggerSpec, error) {
+	var tags [][]generate.TagDefinition
+	var paths []map[string]interface{}
+	var responses []map[string]interface{}
+	var parameters []map[string]interface{}
+	var definitions []map[string]interface{}
+
+	for _, s := range specs {
+		tags = append(tags, s.Tags)
+		paths = append(paths, s.Paths)
+		responses = append(responses, s.Responses)
+		parameters = append(parameters, s.Parameters)
+		definitions = append(definitions, s.Definitions)
+	}
+
+	errs := &multierror.Error{}
+
+	out := generate.SwaggerSpec{
+		Swagger:  common.Swagger,
+		Info:     common.Info,
+		BasePath: common.BasePath,
+		Consumes: common.Consumes,
+		Produces: common.Produces,
+		Schemes:  common.Schemes,
+
+		Tags:        combineTags(common.Tags, tags, errs),
+		Paths:       combineSubSpec(common.Paths, paths, "paths", errs),
+		Responses:   combineSubSpec(common.Responses, responses, "responses", errs),
+		Parameters:  combineSubSpec(common.Parameters, parameters, "parameters", errs),
+		Definitions: combineSubSpec(common.Definitions, definitions, "definitions", errs),
+	}
+
+	return out, errs.ErrorOrNil()
+}
+
+// ConvertToSwagger converts a list of specs and a common spec
+// from YAML format to Swagger structs.
+func ConvertToSwagger(common string, configs []string) ([]generate.SwaggerSpec, generate.SwaggerSpec, error) {
+	specs, err := convertCfgsToSwagger(configs)
+	if err != nil {
+		return nil, generate.SwaggerSpec{}, err
+	}
+
+	commonSpec, err := convertCommonToSwagger(common)
+	if err != nil {
+		return nil, generate.SwaggerSpec{}, err
+	}
+
+	return specs, commonSpec, nil
+}
+
+// convertCfgsToSwagger unmarshals all of the passed in Swagger Specs YAML strings
+// to Swagger structs.
+func convertCfgsToSwagger(configs []string) ([]generate.SwaggerSpec, error){
```

Comment:
Nit: let's keep consistent with referring to these as swagger "specs" rather than mixing "spec" and "config"